### ISSUE TITLE
Unable to populate destination.* for custom span

### ISSUE THREAD

**Issue by tobiasstadler** [elastic/apm-agent-java#1801]
## Is your feature request related to a problem?
We are using a hardware security module (hsm) for doing some crypto stuff. The communication is done via `java.net.Socket`. Currently I use `ElasticApm.currentSpan().startSpan("external", ...)` to create a span for the communication. But the hsm is not listed in the dependencies section of the services overview page in the APM app in Kibana.

## Describe the solution you'd like
I would like to set destination ip/port and destination service name/resource/type for my custom span.

## Describe alternatives you've considered
I could use reflection or write a plugin and use `co.elastic.apm.agent.impl.transaction.Span#getContext` to set the fields, but this looks too over engineered to me.

**Comment by felixbarny** [comment]
Closing as duplicate of #1699
This is currently blocked on a pending spec change: https://github.com/elastic/apm/pull/436

Feel free to chime in on the discussions there
